packages/blocks/src/api/serializer.js

```diff
@@ -255,17 +255,18 @@ export function getCommentDelimitedContent( rawBlockName, attributes, content )
  * Returns the content of a block, including comment delimiters, determining
  * serialized attributes and content form from the current state of the block.
  *
- * @param {Object} block Block instance.
+ * @param {Object} block   Block instance.
+ * @param {Object} options Serialization options.
  *
  * @return {string} Serialized block.
  */
-export function serializeBlock( block ) {
+export function serializeBlock( block, { isNested = false } = {} ) {
 	const blockName = block.name;
 	const saveContent = getBlockContent( block );
 
-	switch ( blockName ) {
-		case getFreeformContentHandlerName():
-		case getUnregisteredTypeHandlerName():
+	switch ( true ) {
```

Comment:
At this point, I think we'd just not want to bother with a `switch` and use `if` instead. Alternatively, we could intentionally use fallthrough to allow the nested case to fall through to `default`.

```js
switch ( blockName ) {
	case getUnregisteredTypeHandlerName():
		return saveContent;

	case getFreeformContentHandlerName():
		if ( ! isNested ) {
			return saveContent;
		}

	default: {
		const blockType = getBlockType( blockName );
		const saveAttributes = getCommentAttributes( blockType, block.attributes );
		return getCommentDelimitedContent( blockName, saveAttributes, saveContent );
	}
}
```

It's pretty confusing though. Maybe less-so if `default` was just code following the `switch` instead of a case, but still confusing I think.